lib/db-utils.js

```diff
@@ -30,41 +44,145 @@ function hashCode(s) {
 }
 
 async function generateId() {
-    return faunaQuery(q.NewId());
+    return faunaQuery(fq.NewId());
 }
 
 function generateLocationId({ name, street, city, zip }) {
     // This deterministically generates a key from the location info.
     const keyString = generateKey({ name, street, city, zip });
     // FaunaDB only accepts positive ref ids, so we make it positive.
     const hash = Math.abs(hashCode(keyString));
-    return hash;
+    return hash.toString();
+}
+
+function addGeneratedIdsToLocations(locations) {
+    return locations.reduce((acc, curr) => {
+        return [
+            {
+                ...curr,
+                // FaunaDB only accepts positive ref ids, so we make it positive.
+                refId: Math.abs(
+                    // hash it
+                    hashCode(
+                        // This deterministically generates a key from the location info.
+                        generateKey({
+                            name: curr.name,
+                            street: curr.street,
+                            city: curr.city,
+                            zip: curr.zip,
+                        })
+                    )
+                ).toString(),
+            },
+            ...acc,
+        ];
+    }, []);
+}
+
+function addScraperRunIdsToLocations(locations) {
+    return locations.reduce(async (acc, curr) => {
+        return [
+            {
+                ...curr,
+                scraperRunId: await generateId(),
+            },
+            ...(await acc),
+        ];
+    }, []);
+}
+
+function addAppointmentsToLocations(locations) {
+    return locations.reduce(async (acc, loc) => {
+        if (loc.hasAvailability && loc.availability) {
+            return [
+                ...(await acc),
+                {
+                    ...loc,
+                    appointments: lodash.compact(
+                        await Promise.all(
+                            Object.entries(loc.availability).map(
+                                async ([date, dateAvailability]) => {
+                                    if (
+                                        dateAvailability.hasAvailability &&
```

Comment:
do we need to worry about the massVax flag, or when we have availability but no signUpLink?

Reply:
Working on this!

Reply:
Added `massVax`, `siteTimestamp`, and `restrictions`. 

I think the frontend filters out entries that have availability but no signUpLink? And since I'm just trying to get the DB behaving like our current setup, I wasn't planning to remove those entries at the moment.

Reply:
Oh, and I only added them to the codepaths we actually call right now (this particular function is part of the batching that we aren't calling at the moment).